Assignment 2/Assignment-2/Validator.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Validator.swift
+//  Assignment-2
+//
+//  Created by Shubham on 26/01/22.
+//
+
+import Foundation
+
+
+func nameValidator(name : String?) ->Bool{
+    if(name == nil){
+        print("wrong name entered")
+        return false
+    }
+    if(name!.isEmpty == true){
+        print("wrong name entered")
+        return false
+    }
+    else{
+        return true
+    }
+}
+
+// Job of this function is to validate whether the input given as string by the users, is valid and contains any of the
+// four madatory courses among A B C D E F
+
+func courseValidation(inputLine : String) -> (Bool){
+    let array = inputLine.components(separatedBy: " ")
+    
+    // checking if same course has been entered twice
+    
+
+    var hash = [String:Int]()
+    for courses in array{
+        if(hash[courses] != nil){
+            return false
+        }
+        else{
+            hash[courses] = 1
+        }
+    }
+    
+    if(array.count != 4){
```

Comment:
Shouldn't the condition here be less than 4 rather than != 4 ?

Reply:
It is mandatory for each student to choose 4 out of 6 courses. This is the statement written in the question. I think the number of courses must be exactly 4.